contracts/TokenConverter/RiskFundConverter.sol

```diff
@@ -213,13 +210,26 @@ contract RiskFundConverter is AbstractTokenConverter {
     /// @param tokenAddress Address of the token
     function getPools(address tokenAddress) internal view returns (address[] memory) {
         address[] memory pools = IPoolRegistry(poolRegistry).getPoolsSupportedByAsset(tokenAddress);
-        uint256 poolsLength = pools.length;
-        address[] memory poolsWithCore = new address[](poolsLength + 1);
 
-        for (uint256 i; i < poolsLength; ++i) {
-            poolsWithCore[i] = pools[i];
+        if (IComptroller(corePoolComptroller).markets(tokenAddress)) {
+            uint256 poolsLength = pools.length;
+            address[] memory poolsWithCore = new address[](poolsLength + 1);
+
+            for (uint256 i; i < poolsLength; ++i) {
+                poolsWithCore[i] = pools[i];
+            }
+            poolsWithCore[poolsLength] = corePoolComptroller;
+            return poolsWithCore;
         }
-        poolsWithCore[poolsLength] = corePoolComptroller;
-        return poolsWithCore;
+
+        return pools;
+    }
+
+    function ensureAssetListed(address comptroller, address asset) internal view returns (bool) {
+        if (comptroller == corePoolComptroller) {
```

Comment:
The keys in the `markets` attribute are vTokens, not underlying tokens

Reply:
https://github.com/VenusProtocol/protocol-reserve/pull/13